### PR TITLE
Release v0.27.0

### PR DESCRIPTION
**Pull request by chukarsten**
# v0.27.0 Jun. 22, 2021
### Enhancements
- Upgraded minimum woodwork to version 0.3.1. Previous versions will not be supported #2181
- Added a new callback parameter for ``explain_predictions_best_worst`` #2308
### Changes
- Deleted the ``return_pandas`` flag from our demo data loaders #2181
- Moved ``default_parameters`` to ``ComponentGraph`` from ``PipelineBase`` #2307
### Documentation Changes
- Updated the release procedure documentation #2230
### Testing Changes
- Ignoring ``test_saving_png_file`` while building conda package #2323
### Breaking Changes
- Deleted the ``return_pandas`` flag from our demo data loaders #2181
- Upgraded minimum woodwork to version 0.3.1. Previous versions will not be supported #2181
- Due to the weak-ref in woodwork, set the result of ``infer_feature_types`` to a variable before accessing woodwork #2181
(base) RM-MB-151:evalml karsten.chu$ tools/format_release_notes.sh 
# v0.27.0 Jun. 22, 2021
### Enhancements
- Adds force plots for prediction explanations #2157
- Removed self-reference from ``AutoMLSearch`` #2304
- Added support for nonlinear pipelines for ``generate_pipeline_code`` #2332
- Added ``inverse_transform`` method to pipelines #2256
- Add optional automatic update checker #2350
- Added ``search_order`` to ``AutoMLSearch``'s ``rankings`` and ``full_rankings`` tables #2345
- Updated threshold optimization method for binary classification #2315
- Updated demos to pull data from S3 instead of including demo data in package #2387
- Upgrade woodwork version to v0.4.1 #2379
### Fixes
- Preserve user-specified woodwork types throughout pipeline fit/predict #2297
- Fixed ``ComponentGraph`` appending target to ``final_component_features`` if there is a component that returns both X and y #2358
- Fixed partial dependence graph method failing on multiclass problems when the class labels are numeric #2372
- Added ``thresholding_objective`` argument to ``AutoMLSearch`` for binary classification problems #2320
- Added change for ``k_neighbors`` parameter in SMOTE Oversamplers to automatically handle small samples #2375
- Changed naming for ``Logistic Regression Classifier`` file #2399
- Pinned pytest-timeout to fix minimum dependence checker #2425
- Replaced ``Elastic Net Classifier`` base class with ``Logistsic Regression`` to avoid ``NaN`` outputs #2420
### Changes
- Cleaned up ``PipelineBase``'s ``component_graph`` and ``_component_graph`` attributes. Updated ``PipelineBase`` ``__repr__`` and added ``__eq__`` for ``ComponentGraph`` #2332
- Added and applied  ``black`` linting package to the EvalML repo in place of ``autopep8`` #2306
- Separated `custom_hyperparameters` from pipelines and added them as an argument to ``AutoMLSearch`` #2317
- Replaced `allowed_pipelines` with `allowed_component_graphs` #2364
- Removed private method ``_compute_features_during_fit`` from ``PipelineBase`` #2359
- Updated ``compute_order`` in ``ComponentGraph`` to be a read-only property #2408
- Unpinned PyZMQ version in requirements.txt #2389 
- Uncapping LightGBM version in requirements.txt #2405
- Updated minimum version of plotly #2415
- Removed ``SensitivityLowAlert`` objective from core objectives #2418
### Documentation Changes
- Fixed lead scoring weights in the demos documentation #2315
- Fixed start page code and description dataset naming discrepancy #2370
### Testing Changes
- Update minimum unit tests to run on all pull requests #2314
- Pass token to authorize uploading of codecov reports #2344
- Add ``pytest-timeout``. All tests that run longer than 6 minutes will fail. #2374
- Separated the dask tests out into separate github action jobs to isolate dask failures. #2376
- Refactored dask tests #2377
- Added the combined dask/non-dask unit tests back and renamed the dask only unit tests. #2382
- Sped up unit tests and split into separate jobs #2365
- Change CI job names, run lint for python 3.9, run nightlies on python 3.8 at 3am EST #2395 #2398
- Set fail-fast to false for CI jobs that run for PRs #2402
### Breaking Changes
- `AutoMLSearch` will accept `allowed_component_graphs` instead of `allowed_pipelines` #2364
- Removed ``PipelineBase``'s ``_component_graph`` attribute. Updated ``PipelineBase`` ``__repr__`` and added ``__eq__`` for ``ComponentGraph`` #2332
- `pipeline_parameters` will no longer accept `skopt.space` variables since hyperparameter ranges will now be specified through `custom_hyperparameters` #2317

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -269,6 +269,7 @@ def setup(app):
     p = Path("/home/docs/.ipython/profile_default/startup")
     p.mkdir(parents=True, exist_ok=True)
     shutil.copy("disable-warnings.py", "/home/docs/.ipython/profile_default/startup/")
+    shutil.copy("set-headers.py", "/home/docs/.ipython/profile_default/startup")
```
Comment (dsherry) [inline docs/source/conf.py:272]: @chukarsten I think something went wrong with your branch?

**Review by freddyaboulton — APPROVED** [review]
Perf tests look good to me @chukarsten ! I think this is good to release.
